ships/0031-shipwright-trigger.md

```diff
@@ -0,0 +1,296 @@
+- <strong><code>.when[].name</code></strong>: short description of what the scenario entails;
+- <strong><code>.when[].type</code></strong>: the type of scenario, namely: <code>Push</code>, <code>PullRequest</code>, <code>Image</code> and <code>Pipeline</code>;
+- <strong><code>.when[].branches[]</code></strong>: slice of branch names for the current repository, branch names where a Build should be activated in case of <code>Push</code> or <code>PullRequest</code> event.
+- <strong><code>.when[].images[]</code></strong>: slice of image names to watch, when those are updated a new Build is activated;
+- <strong><code>.when[].objectRef</code></strong>: used in conjunction with <code>Pipeline</code> (type), and describes a local Tekton Pipeline instance and its current status;
+- <strong><code>.when[].objectRef.name</code></strong>: when informed, the Pipeline is directly search by its name;
+- <strong><code>.when[].objectRef.status[]</code></strong>: slice of Tekton Pipeline [status names](https://tekton.dev/docs/pipelines/pipelineruns/#monitoring-execution-status) where a new Build should be activated. For instance "<em>Succeeded</em>";
+- <strong><code>.when[].objectRef.selector{}</code></strong>: a map of label key-value entries, a alternative to the <code>.objectRef.name</code> attribute, the Tekton Pipeline resource will be identified by label matching;
+
+On the table below, you can see the relationship between the source of the event and the type (<code>.when[].type</code>), described on the <code>Build</code> resource. Please consider:
+
+| Event Source                | Types             | Description                            |
+|-----------------------------|-------------------|----------------------------------------|
+| Git Service Provider        | Push, PullRequest | WebHook events on GitHub, GitLab, etc. |
+| Shipwright Image Controller | Image             | Watching Image objects and reacting    |
+| Tekton Pipeline Controller  | Pipeline          | Watching Tekton objects and reacting   |
+
+All the modifications needed are on the `Build` resource under `.spec.trigger` attribute, making this new feature backwards compatible with `v1alpha1`.
+
+## WebHook Events
+
+The following image represents the actors involved in the WebHook request workflow, and shows the order on which the actors are actionated:
+
+![Shipwright Trigger WebHook Schematics](assets/0031-shipwright-trigger-webhook.drawio.png)
+
+The start of the workflow (1) is at the Shipwright Trigger instance, it will be watching over `Build` objects using a regular Kubernetes Controller. Since the Trigger Controller is aware of all `Build` instances, it creates an inventory of `Builds` carrying trigger rules (`.spec.trigger` attribute), and therefore, it's able to search for `Build` objects. The Trigger will only start listening for WebHook requests when the controller has synchronized, so it has the inventory full beforehand.
```

Comment:
Yeah I was trying to figure out where status was being managed, and then noticed @SaschaSchwarze0 's comment here.

It would not be in this piece of function you are describing here per se, but again, I was not sure where to put my comment, but as it had some synergy with @SaschaSchwarze0 's ....

I was expecting either new conditions or fields in status (I don't have a strong opinion either way) that captured that his BuildRun was "triggered by" either a webhook, image change, pipeline completion, etc.  Analogous to the various trigger by semantics OpenShift Builds has in its BuildRequest:  https://github.com/gabemontero/api/blob/f60a0b2883eae330c01d294b3eba2ee9c3fe7a47/build/v1/types.go#L1236-L1269

Also I'm not seeing anything along those lines in https://github.com/shipwright-io/build/pull/1008/files

Reply:
> Do we really have a controller here = do we reconcile Builds? I did not see a status so far that this new component woud manage. Would an informer cache be what we need so that we know of all existing Build's in the system?

Please consider the [Build Inventory description](https://github.com/otaviof/community/blob/shipwright-trigger-ep/ships/0031-shipwright-trigger.md#build-inventory). Indeed, we have a informer cache prepared to store information in a way we can query later on, to answer which builds are affected by trigger events (WebHook, Kubernetes Event, etc).